crypto/rand/rand_unix.c

```diff
@@ -104,203 +95,111 @@ int RAND_poll(void)
         duration = 1;
         s$sleep(&duration, &code);
 #  else
-#   ifdef OPENSSL_SYS_VOS_IA32
         /* sleep for 1/65536 of a second (15 us).  */
         duration = 1;
         s$sleep2(&duration, &code);
-#   endif                       /* OPENSSL_SYS_VOS_IA32 */
-#  endif                        /* OPENSSL_SYS_VOS_HPPA */
+#  endif
 
-        /* get wall clock time.  */
+        /* get wall clock time, take 8 bits. */
         clock_gettime(CLOCK_REALTIME, &ts);
-
-        /* take 8 bits */
-        v = (unsigned char)(ts.tv_nsec % 256);
+        v = (unsigned char)(ts.tv_nsec & 0xFF);
         RAND_add(&v, sizeof v, 1);
```

Comment:
I didn't notice that restriction.  It might be worthwhile on all machines.

Complex CPUs and operating systems encourage random low order bits without the sleep.  I've found `sleep(0)` to be fairly effective.  It yields the CPU and gives the OS a chance to schedule other tasks.  It does hurt performance if done a lot.